history.py

```diff
@@ -0,0 +1,15 @@
+class LastFiveMiddleware
+    def __init__(self, get_response):
+        self.get_response = get_response
+        # One-time configuration and initialization.
+
+    def __call__(self, request):
+        # Code to be executed for each request before
+        # the view (and later middleware) are called.
+
```

Comment:
#if the session has the last_five variable it will get it, otherwise it will create an empty list
product_ids = request.session.get('last_five', [])
#create the last_five request variable that will be used in the detail.py file
request.last_five = []
#add the products to the request.last_five variable as Product objects, not ids
for p in product_ids:
            request.last_five.append(cmod.Product.objects.get(id=p))